File: ios/ObjC+UIKit/IntegrationDemo_ObjC_UIKit/PlacementCell.m
//
//  PlacementCell.m
//  IntegrationDemo_ObjC_UIKit
//
//  Created by Jeroen Verbeek on 2/22/23.
//

#import "PlacementCell.h"
#import <TapResearchSDK/TapResearchSDK.h>

@interface PlacementCell ()

@property (weak, nonatomic) IBOutlet UILabel *title;
@property (weak, nonatomic) IBOutlet UILabel *subLabel;

@end

@implementation PlacementCell

+ (PlacementCell*)cellForTableView:(UITableView*)tableView placementTag:(NSString*)tag andInfo:(NSString*)info {

	static NSString *cellId = @"PlacementCell";
	PlacementCell *cell = (PlacementCell *)[tableView dequeueReusableCellWithIdentifier:cellId];
	if (!cell) {
		cell = [[PlacementCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellId];
	}
	[cell fillCellWithPlacementTag:tag andInfo:info];
	return cell;
}

- (void)fillCellWithPlacementTag:(NSString*)placement andInfo:(NSString*)info {

	self.accessibilityLabel = placement;
	self.title.text = placement;
	if (info) {
		self.subLabel.text = info;
	}
	else {
		TRPlacementDetails *details = [TapResearch getPlacementDetails:placement errorHandler:^(NSError * _Nullable error) {
		}];
		if (details) {
			self.subLabel.text = [NSString stringWithFormat:@"%f, %@, ends: %@", details.saleMultiplier, details.saleDisplayName, details.saleEndDate];
		}
		else {
			self.subLabel.text = @"";
		}
	}
}

@end

File: ios/Swift+UIKit/IntegrationDemo_Swift_UIKit/PlacementCell.swift
//
//  PlacementCell.swift
//  IntegrationDemo_Swift_UIKit
//
//  Created by Jeroen Verbeek on 3/16/23.
//

import UIKit

class PlacementCell : UITableViewCell {

	@IBOutlet weak var title: UILabel!
	@IBOutlet weak var subLabel: UILabel!

	class func cell(tableView: UITableView, placement: String, info: String? = nil) -> PlacementCell {

		var cell: PlacementCell? = tableView.dequeueReusableCell(withIdentifier: "PlacementCell") as? PlacementCell
		if(cell == nil) {
			cell = PlacementCell(style: .default, reuseIdentifier: "PlacementCell")
		}
		cell?.fillCell(placement: placement, info: info)
		return cell!
	}

	func fillCell(placement: String, info: String?) {

		self.accessibilityLabel = placement
		title.text = placement
		if let info = info {
			subLabel.text = info
		}
		else {
			subLabel.text = nil
		}
	}

}
